xen/arch/arm/coproc/coproc.c

```diff
@@ -183,9 +183,19 @@ static int coproc_attach_to_domain(struct domain *d,
         goto out;
     }
 
+    ret = iommu_assign_coproc(d, coproc->dev);
```

Comment:
I guess the check if this coproc needs IOMMU at all is done inside iommu_assign_coproc and if this is not the case no error returned?